File: ancillary/RDemeter/RDemeter_draft.py
# Marcus Perry, PNNL 03/24/23
# draft python script using demo data from Demeter to run Demeter within R using R Reticulate package.

# pip install h5py, netCDF4 # (if not already installed)

import demeter

def run_demeterR(config_file_path, write_outputs_flag):
  config_file = config_file_path
  write_outputs = write_outputs_flag
  demeter.run_model(config_file=config_file, write_outputs=write_outputs)
  

File: ancillary/RDemeter/RDemeter_draft.R
## Marcus Perry, PNNL/JCGRI 03/23/23
## R script for running Demeter in R

# install.packages("reticulate") # Do the first time
library(reticulate)

source_python("RDemeter.py")
# now python function can be called() as normal within R body
config_file_path = '~\\config_gcam_reference.ini' # your path here
write_outputs_flag = TRUE
# write_log_flag=FALSE # Maybe add in later
run_demeterR(config_file_path, write_outputs_flag)
